crates/circuit/src/dag_circuit.rs

```diff
@@ -4491,40 +4543,61 @@ impl DAGCircuit {
             .len()
     }
 
+    /// Total number of stretches tracked by the circuit.
+    #[getter]
+    fn num_stretches(&self, py: Python) -> usize {
+        self.num_captured_stretches(py) + self.num_declared_stretches(py)
+    }
+
+    /// Number of captured stretches tracked by the circuit.
+    #[getter]
+    fn num_captured_stretches(&self, py: Python) -> usize {
+        self.captured_stretches.bind(py).len()
+    }
+
+    /// Number of declared local stretches tracked by the circuit.
+    #[getter]
+    fn num_declared_stretches(&self, py: Python) -> usize {
+        self.declared_stretches.bind(py).len()
+    }
```

Comment:
If this is for 2.0 I don't think we can backport this since these are new attributes for the dag. This would only be for 2.1. They make total sense to add, but we might want to split this PR out into multiple parts.

Reply:
Makes sense. I've removed the backport label and will add a release note for this PR for 2.1.